src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -1126,6 +1133,11 @@ public function move($blnIsAjax=false)
 			// Redirect or reload
 			if (!$objUploader->hasError())
 			{
+				if ($blnIsAjax)
+				{
+					throw new ResponseException(new Response(\Message::generateUnwrapped(), 201));
```

Comment:
IMO 204 is out of question, as we are returning a response containing `\Message::generateUnwrapped()` and therefore have response content.

For the `Location` header, please see [rfc7231 section 6.3.2](https://tools.ietf.org/html/rfc7231#section-6.3.2) (which obsoletes your referenced RFC 2616):

> The 201 (Created) status code indicates that the request has been fulfilled and has resulted in one or more new resources being created.  The primary resource created by the request is identified by **either a Location header field in the response or, if no Location field is received, by the effective request URI.**
> The 201 response payload typically describes and links to the resource(s) created.

Therefore we **MUST** either send a `Location` header OR the URI it got posted at is effectively the correct one. The standard has been made more strict.

However, I see a solution by sending a `Location` header for the first uploaded file and `Link` headers (see RFC [rfc5988 section 5](https://tools.ietf.org/html/rfc5988#section-5)) for any subsequent file in the original payload.